samples/common/src/utils.rs

```diff
@@ -96,26 +96,15 @@ pub async fn discover_digital_twin_provider_using_ibeji(
 
     let entity_access_info = response_inner.entity_access_info.expect("Did not find the entity");
 
-    let mut matching_endpoint_info_option: Option<EndpointInfo> = None;
-    for endpoint_info in entity_access_info.endpoint_info_list {
-        // We require and endpoint that supports the protocol and supports all of the operations.
-        if endpoint_info.protocol == protocol
-            && is_subset(operations, endpoint_info.operations.as_slice())
-        {
-            matching_endpoint_info_option = Some(endpoint_info);
-            break;
-        }
-    }
-
-    if matching_endpoint_info_option.is_none() {
-        return Err("Did not find an endpoint that met our requirements".to_string());
+    match entity_access_info.endpoint_info_list.iter()
```

Comment:
instead of a match statement here you could consider using `ok_or`, though that would not let you include the info statement on line 103 easily if you want to have that